### ISSUE TITLE
no-undef misleading example

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#11963]
**The version of ESLint you are using.**

master

**The problem you want to solve.**

From `no-undef` docs:

> Examples of **correct** code for this rule with `global` declaration:
>
> ```js
> /*global someFunction b:writable*/
> /*eslint no-undef: "error"*/
>
> var a = someFunction();
> b = 10;
> ```
>
> The `b:writable` syntax in `/*global */` indicates that assignment to `b` is correct.
>
> Examples of **incorrect** code for this rule with `global` declaration:
> 
> ```js
> /*global b*/
> /*eslint no-undef: "error"*/
> 
> b = 10;
> ```
> 
> By default, variables declared in `/*global */` are read-only, therefore assignment is incorrect.

'Incorrect' examples should be those with warnings, but ESLint does not report any warnings for the second example.

The problem with this is that the second example is actually **correct**, because `no-undef` rule does not check assignments to readonly global variables, that's what `no-global-assign` does.

**Your take on the correct solution to problem.**

I'm not sure. Probably modify the first example and remove the second. Maybe reference `no-global-assign`.  Perhaps also add more info to  `no-global-assign`. 

**Are you willing to submit a pull request to implement this change?**
Yes.

**Comment by platinumazure** [comment]
Hi @mdjermanovic, thanks for the issue.

I think `no-undef` used to handle this case in the past, so it makes sense that the documentation might now incorrectly claim that the rule will check global assignments. We would be grateful for a PR that fixes this.

**Comment by mdjermanovic** [comment]
The interesting this here is that `no-implicit-globals` also reports this, albeit in a different context.

```js
/* eslint no-implicit-globals:"error" */
/* eslint no-global-assign:"error" */
/* global a:readonly */

a = 1;
```

```
  5:1  error  Implicit global variable, assign as global property instead  no-implicit-globals
  5:1  error  Read-only global 'a' should not be modified                  no-global-assign
```

None of the rules will report anything if the global variable is set to `writeable`.

Is this the intended behavior of `no-implicit-globals`?

**Comment by mysticatea** [comment]
I guess that it's intended because [there are tests](https://github.com/eslint/eslint/commit/48ee9997a4999282c2cca671e6f77a72ce1d2cc8#diff-1db90d89fc8ce3abee7753642cbc7fcdR281).

But it sounds a bug to me. In the original issue #4542, they talked about declarations.
